src/main/java/com/dev/cinema/dao/impl/ShoppingCartDaoImpl.java

```diff
@@ -0,0 +1,70 @@
+package com.dev.cinema.dao.impl;
+
+import com.dev.cinema.dao.ShoppingCartDao;
+import com.dev.cinema.exception.DataProcessingException;
+import com.dev.cinema.lib.Dao;
+import com.dev.cinema.model.ShoppingCart;
+import com.dev.cinema.model.User;
+import com.dev.cinema.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class ShoppingCartDaoImpl implements ShoppingCartDao {
+    @Override
+    public ShoppingCart add(ShoppingCart shoppingCart) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(shoppingCart);
+            transaction.commit();
+            return shoppingCart;
+        } catch (Exception e) {
+            if (transaction != null && transaction.isActive()) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't create shopping cart " + shoppingCart, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<ShoppingCart> query = session.createQuery("from ShoppingCart sc "
+                    + "left join fetch sc.tickets "
+                    + "where sc.user = :user", ShoppingCart.class);
+            query.setParameter("user", user);
+            return query.getSingleResult();
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get from db by this user " + user, e);
+        }
+    }
+
+    @Override
+    public void update(ShoppingCart shoppingCart) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.update(shoppingCart);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null && transaction.isActive()) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't create shopping cart " + shoppingCart, e);
```

Comment:
```suggestion
            throw new DataProcessingException("Can't update shopping cart " + shoppingCart, e);
```
I guess it would be better =)

Reply:
agree )